VTUL/dlp-access
# Refactor image grid to show images for collection and it's children

**Refactor image grid to show images for collection and it's children.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2255) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Refactor image grid to show images for collection and it's children.

# What's the changes? (:star:)
* Adds `hierarchy_path` field to Archive table so that each archive will know all of the collections that it belongs to
* Updates query in CollectionItemsLoader.js to include the archives of a collections children as well.


# How should this be tested?
* Changes can be viewed here: https://libtd-2255.djf1n0m63xbku.amplifyapp.com/
* Visit several collection show pages and check that image grid contains archives that belong to the current collection OR any of it's children.

# Additional Notes:
* branch: `LIBTD-2255`
* After this PR is merged and the data model changes are pushed to the dev amplify environment, I think it would probably be easiest to use the DDBtoDDB tool to copy all three tables over from the `leearchive` environment to the `dev` environment.
* In the future when new items are added, the `archiveCollectionHeirarchy` lambda function needs to be run.

# Interested parties
@yinlinchen 

(:star:) Required fields
